mlflow/projects/backend/abstract_backend.py

```diff
@@ -0,0 +1,21 @@
+from abc import abstractproperty, abstractmethod, ABCMeta
+
+
+class AbstractBackend():
+    """
+    Abstract class for MLflow Project Execution Backend
+    A backend must implement the run method to launch a new job
+    """
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def run(self, active_run, uri, entry_point, work_dir, parameters,
```

Comment:
Would you mind adding a docstring for this method describing the params & their types? 

Also, it'd be good to write out the possible options for this method signature & the pros/cons of the different options, e.g. another option could be to accept **kwargs here, but we might prefer that users just specify custom backend-specific arguments via `cluster_spec` (which BTW I think should be renamed to `backend_config`?)

It'd also be good to write out a high-level description of how this interface would work for the kubernetes & local backends, given that it excludes arguments like `storage_dir` that are passed to the local backend (maybe we just special-case such arguments & handle them before delegating to the plugin?

Reply:
done